### PR TITLE
fix(tests): pick up latest auth db in auth server tests

### LINKED ISSUE #793: Auth server package.json still depends on github.com/mozilla/fxa-auth-db-mysql
As per https://github.com/mozilla/fxa/pull/791#issuecomment-481640616, the auth server still points at the old db repo for running the tests against, so it's not going to pick up any new changes. We should fix that.

### PR DESCRIPTION
**Pull request by philbooth**
Fixes #793.

This is what I had in mind to hook the auth server tests up with the auth db from this repo. When Circle is back I'll restart the build to see how it gets on.

### CODE REVIEW

**Comment by philbooth** [comment]
This got close to where I wanted it to be, but some remote tests are failing and at least some of those failures look like they're related to the same unicode MySQL stuff I suffered with in the email service (ref #782):

```
  299 passing (5m)
  10 failing

  1) remote account create
       account creation works with unicode email address:
     An internal server error occurred
  

  2) remote account login
       login works with unicode email address:
     An internal server error occurred
  

  3) remote account profile
       account profile works with unicode email address:
     An internal server error occurred
  

  4) remote account status
       account status by email works with unicode email address:
     An internal server error occurred
  

  5) remote base path
       "/" returns valid version information:
     AssertionError: source repository: expected false to be truthy
      at request.spread (test/remote/base_path_tests.js:34:18)
      at tryCatcher (node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (node_modules/bluebird/js/release/promise.js:509:35)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:569:18)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:694:18)
      at Promise._fulfill (node_modules/bluebird/js/release/promise.js:638:18)
      at PromiseArray._resolve (node_modules/bluebird/js/release/promise_array.js:126:19)
      at PromiseArray._promiseFulfilled (node_modules/bluebird/js/release/promise_array.js:144:14)
      at PromiseArray._iterate (node_modules/bluebird/js/release/promise_array.js:114:31)
      at PromiseArray.init [as _init] (node_modules/bluebird/js/release/promise_array.js:78:10)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:566:21)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:694:18)
      at Promise._fulfill (node_modules/bluebird/js/release/promise.js:638:18)
      at Request._callback (node_modules/bluebird/js/release/nodeback.js:45:21)
      at Request.self.callback (node_modules/request/request.js:185:22)
      at Request.EventEmitter.emit (domain.js:441:20)
      at Request.<anonymous> (node_modules/request/request.js:1161:10)
      at Request.EventEmitter.emit (domain.js:441:20)
      at IncomingMessage.<anonymous> (node_modules/request/request.js:1083:12)
      at IncomingMessage.EventEmitter.emit (domain.js:441:20)
      at endReadableNT (_stream_readable.js:1125:12)
      at process._tickCallback (internal/process/next_tick.js:63:19)

  6) remote base path
       "/__version__" returns valid version information:
     AssertionError: source repository: expected false to be truthy
      at request.spread (test/remote/base_path_tests.js:34:18)
      at tryCatcher (node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (node_modules/bluebird/js/release/promise.js:509:35)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:569:18)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:694:18)
      at Promise._fulfill (node_modules/bluebird/js/release/promise.js:638:18)
      at PromiseArray._resolve (node_modules/bluebird/js/release/promise_array.js:126:19)
      at PromiseArray._promiseFulfilled (node_modules/bluebird/js/release/promise_array.js:144:14)
      at PromiseArray._iterate (node_modules/bluebird/js/release/promise_array.js:114:31)
      at PromiseArray.init [as _init] (node_modules/bluebird/js/release/promise_array.js:78:10)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:566:21)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:694:18)
      at Promise._fulfill (node_modules/bluebird/js/release/promise.js:638:18)
      at Request._callback (node_modules/bluebird/js/release/nodeback.js:45:21)
      at Request.self.callback (node_modules/request/request.js:185:22)
      at Request.EventEmitter.emit (domain.js:441:20)
      at Request.<anonymous> (node_modules/request/request.js:1161:10)
      at Request.EventEmitter.emit (domain.js:441:20)
      at IncomingMessage.<anonymous> (node_modules/request/request.js:1083:12)
      at IncomingMessage.EventEmitter.emit (domain.js:441:20)
      at endReadableNT (_stream_readable.js:1125:12)
      at process._tickCallback (internal/process/next_tick.js:63:19)

  7) remote email validity
       /account/create with a variety of unusual but valid email addresses:
     AssertionError: Email address Ӓ@example.com should have been allowed, but it wasn't
      at emails.(anonymous function).Client.create.then (test/remote/email_validity_tests.js:76:15)
      at tryCatcher (node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (node_modules/bluebird/js/release/promise.js:512:31)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:569:18)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:690:18)
      at _drainQueueStep (node_modules/bluebird/js/release/async.js:138:12)
      at _drainQueue (node_modules/bluebird/js/release/async.js:131:9)
      at Async._drainQueues (node_modules/bluebird/js/release/async.js:147:5)
      at Immediate.Async.drainQueues [as _onImmediate] (node_modules/bluebird/js/release/async.js:17:14)
      at process.topLevelDomainCallback (domain.js:120:23)

  8) remote misc
       / returns version, git hash and source repo:
     AssertionError: source repository: expected false to be truthy
      at request.spread (test/remote/misc_tests.js:32:16)
      at tryCatcher (node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (node_modules/bluebird/js/release/promise.js:509:35)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:569:18)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:694:18)
      at Promise._fulfill (node_modules/bluebird/js/release/promise.js:638:18)
      at PromiseArray._resolve (node_modules/bluebird/js/release/promise_array.js:126:19)
      at PromiseArray._promiseFulfilled (node_modules/bluebird/js/release/promise_array.js:144:14)
      at PromiseArray._iterate (node_modules/bluebird/js/release/promise_array.js:114:31)
      at PromiseArray.init [as _init] (node_modules/bluebird/js/release/promise_array.js:78:10)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:566:21)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:694:18)
      at Promise._fulfill (node_modules/bluebird/js/release/promise.js:638:18)
      at Request._callback (node_modules/bluebird/js/release/nodeback.js:45:21)
      at Request.self.callback (node_modules/request/request.js:185:22)
      at Request.EventEmitter.emit (domain.js:441:20)
      at Request.<anonymous> (node_modules/request/request.js:1161:10)
      at Request.EventEmitter.emit (domain.js:441:20)
      at IncomingMessage.<anonymous> (node_modules/request/request.js:1083:12)
      at IncomingMessage.EventEmitter.emit (domain.js:441:20)
      at endReadableNT (_stream_readable.js:1125:12)
      at process._tickCallback (internal/process/next_tick.js:63:19)

  9) remote misc
       /__version__ returns version, git hash and source repo:
     AssertionError: source repository: expected false to be truthy
      at request.spread (test/remote/misc_tests.js:32:16)
      at tryCatcher (node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (node_modules/bluebird/js/release/promise.js:509:35)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:569:18)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:694:18)
      at Promise._fulfill (node_modules/bluebird/js/release/promise.js:638:18)
      at PromiseArray._resolve (node_modules/bluebird/js/release/promise_array.js:126:19)
      at PromiseArray._promiseFulfilled (node_modules/bluebird/js/release/promise_array.js:144:14)
      at PromiseArray._iterate (node_modules/bluebird/js/release/promise_array.js:114:31)
      at PromiseArray.init [as _init] (node_modules/bluebird/js/release/promise_array.js:78:10)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:566:21)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:694:18)
      at Promise._fulfill (node_modules/bluebird/js/release/promise.js:638:18)
      at Request._callback (node_modules/bluebird/js/release/nodeback.js:45:21)
      at Request.self.callback (node_modules/request/request.js:185:22)
      at Request.EventEmitter.emit (domain.js:441:20)
      at Request.<anonymous> (node_modules/request/request.js:1161:10)
      at Request.EventEmitter.emit (domain.js:441:20)
      at IncomingMessage.<anonymous> (node_modules/request/request.js:1083:12)
      at IncomingMessage.EventEmitter.emit (domain.js:441:20)
      at endReadableNT (_stream_readable.js:1125:12)
      at process._tickCallback (internal/process/next_tick.js:63:19)

  10) remote password forgot
       forgot password:
     AssertionError: flow begin time set: expected undefined to equal 1554983557621
      at Client.createAndVerify.then.then.then.then.then.then.then (test/remote/password_forgot_tests.js:87:20)
      at tryCatcher (node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (node_modules/bluebird/js/release/promise.js:512:31)
      at Promise._settlePromise (node_modules/bluebird/js/release/promise.js:569:18)
      at Promise._settlePromise0 (node_modules/bluebird/js/release/promise.js:614:10)
      at Promise._settlePromises (node_modules/bluebird/js/release/promise.js:694:18)
      at _drainQueueStep (node_modules/bluebird/js/release/async.js:138:12)
      at _drainQueue (node_modules/bluebird/js/release/async.js:131:9)
      at Async._drainQueues (node_modules/bluebird/js/release/async.js:147:5)
      at Immediate.Async.drainQueues [as _onImmediate] (node_modules/bluebird/js/release/async.js:17:14)
      at process.topLevelDomainCallback (domain.js:120:23)
```

Obviously we can't just delete the failing tests here like I did for the email service, so I'm tweaking the approach to make sure it still uses `Dockerflow-test` like it did before. If it works that might be the correct path to take in #782 as well.

**Comment by philbooth** [comment]
Okay, I think this is working now.

The approach I've taken is to `require` a local copy of `fxa-auth-db-mysql` in the tests, then make sure that is available in most contexts:

* For local dev, `_scripts/install_all.sh` copies it so it should be there for anyone who sets up the monorepo from scratch. (and I took the liberty of doing something similar for `fxa-email-service`)

* There's also a belt-and-braces check in `fxa-auth-server/scripts/test-local.sh` that copies it if it can, but is non-fatal if run in a context where it doesn't exist to copy from. This is so that it works for people who haven't done an `npm i` from the monorepo root yet.

* For CI, it's been added to `fxa-auth-server/Dockerfile-test` with a commensurate step added to `.circleci/build.sh` that makes the copy available when building the Docker image.

However, there is nothing that copies the directory for `fxa-dev`. I didn't realise before but those images don't even have the dev dependencies (sorry @vbudhram, you did try to point this out to me in https://github.com/mozilla/fxa/pull/778#discussion_r273533547, but I didn't realise the difference between the two Dockerfiles at that point).

The core logic for copying/running the auth db has been pulled out to `_scripts/clone-authdb.sh` since it is used in so many places.

@mozilla/fxa-devs r?

**Comment by philbooth** [comment]
@mozilla/fxa-devs, this PR always uses the `master` branch for the copy of `fxa-auth-db-mysql`. Would it be better to use the current `HEAD` instead? That should mean it works for pull requests and `master` and CI...

**Comment by philbooth** [comment]
> Would it be better to use the current `HEAD` instead?

Fwiw I think that makes heaps more sense, plus it also avoids all that silly git `sparse-checkout` malarkey I had going on in the script because it can just copy from the local file system now. Not sure why I didn't just start off like that tbh. 😕

Anyway, have pushed with that change.

File: servers.json [hunk 1]
```diff
@@ -12,8 +12,8 @@
     },
     {
       "name": "auth-server db memory PORT 8000",
-      "script": "node_modules/fxa-auth-db-mysql/bin/mem.js",
```
Comment (vbudhram) [inline servers.json]: Nice catch! Completely forgot about this.

**Comment by vbudhram** [comment]
> Would it be better to use the current HEAD instead?

Long term I think we should do that, but probably ok to get this in and iterate scripts.

**Pull request merged**